From 90533488e3e923f65a24fe94a055477c2ab4b8d1 Mon Sep 17 00:00:00 2001
From: milde <milde@929543f6-e4f2-0310-98a6-ba3bd3dd1d04>
Date: Tue, 3 Mar 2020 20:32:40 +0000
Subject: [PATCH] Fix tests of syntax highlighting with Pygments >=2.5.

Pygments changed the default Python lexer from "python2"
to "python3" which results in classification of
"print" as "name builtin" instead of "keyword".

Specifying "python3" in the test samples ensures the same output
with Pygments 2.4 and 2.5.

git-svn-id: http://svn.code.sf.net/p/docutils/code/trunk@8495 929543f6-e4f2-0310-98a6-ba3bd3dd1d04
---
 .../test/test_parsers/test_rst/test_directives/test_code.py  | 12 ++++++------
 .../test_parsers/test_rst/test_directives/test_code_long.py  |  6 +++---
 docutils/test/test_parsers/test_rst/test_interpreted.py      |  6 +++---
 3 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/test/test_parsers/test_rst/test_directives/test_code.py b/test/test_parsers/test_rst/test_directives/test_code.py
index c9ee2c014..c68df0542 100644
--- a/test/test_parsers/test_rst/test_directives/test_code.py
+++ b/test/test_parsers/test_rst/test_directives/test_code.py
@@ -98,16 +98,16 @@ totest['code'] = [
 
 totest['code-parsing'] = [
 ["""\
-.. code:: python
+.. code:: python3
   :class: testclass
 
    print 'hello world' # to stdout
 """,
 """\
 <document source="test data">
-    <literal_block classes="code python testclass" xml:space="preserve">
+    <literal_block classes="code python3 testclass" xml:space="preserve">
          \n\
-        <inline classes="keyword">
+        <inline classes="name builtin">
             print
         <inline classes="punctuation">
             (
@@ -120,7 +120,7 @@ totest['code-parsing'] = [
             # to stdout
 """],
 ["""\
-.. code:: python
+.. code:: python3
   :class: testclass
   :name: my_function
   :number-lines: 7
@@ -134,7 +134,7 @@ totest['code-parsing'] = [
 """,
 """\
 <document source="test data">
-    <literal_block classes="code python testclass" ids="my-function" names="my_function" xml:space="preserve">
+    <literal_block classes="code python3 testclass" ids="my-function" names="my_function" xml:space="preserve">
         <inline classes="ln">
              7 \n\
         <inline classes="keyword">
@@ -167,7 +167,7 @@ totest['code-parsing'] = [
         <inline classes="ln">
             12 \n\
             \n\
-        <inline classes="keyword">
+        <inline classes="name builtin">
             print
         <inline classes="punctuation">
             (
diff --git a/test/test_parsers/test_rst/test_directives/test_code_long.py b/test/test_parsers/test_rst/test_directives/test_code_long.py
index 22541f534..89249f73f 100644
--- a/test/test_parsers/test_rst/test_directives/test_code_long.py
+++ b/test/test_parsers/test_rst/test_directives/test_code_long.py
@@ -24,7 +24,7 @@ totest = {}
 
 totest['code-parsing-long'] = [
 ["""\
-.. code:: python
+.. code:: python3
   :number-lines: 7
 
   def my_function():
@@ -36,7 +36,7 @@ totest['code-parsing-long'] = [
 """,
 """\
 <document source="test data">
-    <literal_block classes="code python" xml:space="preserve">
+    <literal_block classes="code python3" xml:space="preserve">
         <inline classes="ln">
              7 \n\
         <inline classes="keyword">
@@ -69,7 +69,7 @@ totest['code-parsing-long'] = [
         <inline classes="ln">
             12 \n\
             \n\
-        <inline classes="keyword">
+        <inline classes="name builtin">
             print
         <inline classes="punctuation">
             (
diff --git a/test/test_parsers/test_rst/test_interpreted.py b/test/test_parsers/test_rst/test_interpreted.py
index 8ec44bc30..2727a8a3c 100755
--- a/test/test_parsers/test_rst/test_interpreted.py
+++ b/test/test_parsers/test_rst/test_interpreted.py
@@ -271,7 +271,7 @@ Custom role based on code role:
 Custom role based on code role:
 
 .. role:: python(code)
-   :language: python
+   :language: python3
    :class: testclass
 
 Python code :python:`print("The end")`.
@@ -282,8 +282,8 @@ Python code :python:`print("The end")`.
         Custom role based on code role:
     <paragraph>
         Python code \n\
-        <literal classes="code testclass python">
-            <inline classes="keyword">
+        <literal classes="code testclass python3">
+            <inline classes="name builtin">
                 print
             <inline classes="punctuation">
                 (
-- 
2.11.4.GIT

